jvm/library/src/main/kotlin/io/tira/tirex/tracker/Tracker.kt

```diff
@@ -671,11 +672,37 @@ class TrackingHandle private constructor(
     private fun export(result: Pointer) {
         if (exportFilePath == null) return
         when (exportFormat) {
-            null -> return
+            null -> exportGuessedFormat(result)
             ExportFormat.IR_METADATA -> exportIrMetadata(result)
         }
     }
 
+    private fun exportGuessedFormat(result: Pointer) {
+        if (exportFilePath == null) return
+        else if (
+            listOf(
+                "ir_metadata",
+                "ir-metadata",
+                "irmetadata",
+                "ir_metadata.yml",
+                "ir-metadata.yml",
+                "irmetadata.yml",
+                "ir_metadata.yaml",
+                "ir-metadata.yaml",
+                "irmetadata.yaml",
+                "ir_metadata.gz",
+                "ir-metadata.gz",
+                "irmetadata.gz",
+                "ir_metadata.yml.gz",
+                "ir-metadata.yml.gz",
+                "irmetadata.yml.gz",
+                "ir_metadata.yaml.gz",
+                "ir-metadata.yaml.gz",
+                "irmetadata.yaml.gz",
+            ).any { exportFilePath.name.endsWith(it) }
```

Comment:
Would a regex be prettier here?
`ir[-_]?metadata(.ya?ml)?(.gz)?`
The regex could also be compiled to be case insensitive.

Reply:
I'd rather keep it explicit and also require explicitly lower-case extension for auto-guessing the format. Users who want to use some other non-standard file extension can still override the format with the `export_format` flag.